### DISCUSSION TITLE
connessione di sensori in modalità remota via ddns

### DISCUSSION THREAD

**Discussion by marioroma** [indomus/forum#2177] [Networking]
Buon giorno a tutti
Sono alle prime armi con HA, prima ho provato vedere se l'argomento era stato toccato ma non sono riuscito a trovar nulla
Vorrei collegare al mio HA nella rete di casa (router con schedina telefonica quindi dietro NAT) alcuni sensori che ho in una località diversa montati su esp32 e esp8266
Mi collego a internet attraverso un router terzo (della società che mi ospita) sempre dietro NAT
Attraverso il DDNS di Duck ho accesso remoto al mio server con un collegamento tipo http: xxxxxxx.duckdns.org:8123
Le schede esp sono configurate per accedere al server con un indirizzo IP sulla stessa rete
Non potendo, per ovvi motivi, creare una VPN tra i due router  mi domandavo se c'era un qualche modo per dire alla scheda di cercare il server non con un IP ma con l'URL con cui mi collego col terminale di HA.
Spero di esser stato in grado di spiegare la mia esigenza
Ringrazio intanto chi mi darà una mano o cercherà di farlo

**Comment by snakuzzo** [comment]
Credo/spero di aver capito.
Innanzi tutto un consiglio...cerca di esporre HA in https e non in http.
Quanto alla tua esigenza, non è il device a commettersi ad HA, ma l'esatto contrario... è HA a dover "vedere" i device, a meno che tu non decida di utilizzare MQTT come protocollo.bpoi dipende tutto anche dal firmware che hai utilizzato sui device (tasmota? esphome?)

In linea di massima...

Se usi MQTT sul tuo server di casa dovresti fare in modo di esporlo su internet e poi i tuoi device si collegherebbero ad esso.

Se non usi MQTT, probabilmente userai le API HTTP. Qui dovresti fare in modo di esporre su internet i device e configurare su HA le varie entità utilizzando le API (Se usassi esphome ti basterebbe esporre i device su internet e poi da HA aggiungerli con IP e porta con integrazione esphome).

Al momento non mi viene in mente altro.

**Reply by marioroma** [reply]
Sì, scusa, avrei dovuto chiarirlo: esphome
Http perchè non son riuscito ad attivare anche https: al momento può andar bene così, niente di critico, poi vedrò di attivare https
Ho caricato il server Mqtt ma non lo sto esplicitamente utilizzando su queste schede
Siccome anche il router esterno a cui si collegano le schede è un router con SIM ( dietro NAT ) non saprei se si può, e poi come fare, a esporle su internet
Se mi dici che si può vado a cercare le relative info operative
Intanto grazieK per le tue considerazioni

**Comment by 50494554524F** [comment]
(non provato, idea se non riesci con vpn) se puoi avere home assistant in entrambi i posti, usa l'addon tor (roba da due minuti) e poi connetti le due istanze con remote-homeassistant che trovi in hacs

(non provato anche questo, se hai una sola instanza di homeassistant) eventualmente potresti impostare esphome per chiamare una url duckdns a cui appendere i valori dei sensori e che leggerai dalla tua instanza di homeassistant

**Reply by snakuzzo** [reply]
Il primo consiglio, non credo faccia al caso suo
il secondo, è sostanzialmente (ma scritto non proprio bene) il consiglio che gli ho già dato io

Infine, ti chiedo cortesemente di non creare nuovi thread, rispondi sotto l'ultimo reply. Grazie

**Reply by marioroma** [reply]
Intendi usando questa finestra?
Ok, scusate
Se non riesco altrimenti pensavo di prendere un pi3 che gestisca quei quattro sensori e li passi a quello di casa.
VPN esclusa al momento
Proverò una simulazione di collegamento (quando capirò cosa fare) da casa tramite il AP di un cellulare
Al momento sono incastrato con un passaggio che non mi riesce e non voglio rompere oltremisura

**Reply by snakuzzo** [reply]
non dicevo a te :D
Comunque, se i device vanno su internet tramite una connessione da cellulare, al 99% sarai nattato e quindi non sarebbero direttamente raggiungibili da internet.

Secondo me la soluzione migliore sarebbe:

1. esponi un broker MQTT su internet (da casa tua...quella dove riesci anche a raggiungere HA) in modalità sicura
2. Flashi il firmware dei device esphome disabilitando abilitando mqtt e disabilitando le api ... altrimenti si riavvierebbe ogni tot minuti
3. Colleghi i device al tuo broker

**Reply by marioroma** [reply]
Comunque anche il router dove si trovano i sensori è, essendo collegato con una sim mobile, dietro NAT.
Ricapitolando : devo fare di nuovo l'install alle schede dopo aver modificato il loro file togliendo le API e inserendo la configurazione per accere a Mqtt
Con "Colleghi i device al tuo broker" intendi configurare l'accesso al Mqtt  indicando http(s)://xxxxxx.duckdns.org:1883  al posto del suo IP? 
E' possibile?

**Reply by snakuzzo** [reply]
In linea di massima sì.
Ma la mia è solo una delle possibili opzioni...io proverei. Non ti ho detto "devi fare così", né ti posso dire esattamente cosa devi fare.

Poi lato HA dovresti creare tutte entità MQTT a mano.
Altra cosa importante, la connessione sicura verso casa tua (per HA e MQTT).

**Reply by marioroma** [reply]
grazie
